tests/macsec/macsec_helper.py

```diff
@@ -472,18 +472,104 @@ def macsec_dp_poll(test, device_number=0, port_number=None, timeout=None, exp_pk
+            -------------------------------------  ----------------------------------------------------------------
+            auth_key                               XXX
+            next_pn                                1
+            sak                                    XXX
+            salt                                   XXX
+            ssci                                   2
+            SAI_MACSEC_SA_ATTR_CURRENT_XPN         8
+            SAI_MACSEC_SA_STAT_OCTETS_ENCRYPTED    28532
+            SAI_MACSEC_SA_STAT_OCTETS_PROTECTED    0
+            SAI_MACSEC_SA_STAT_OUT_PKTS_ENCRYPTED  7
+            SAI_MACSEC_SA_STAT_OUT_PKTS_PROTECTED  0
+            -------------------------------------  ----------------------------------------------------------------
+            MACsec Ingress SC (XXX)
+
+            MACsec Ingress SA (1)
+            ---------------------------------------  ----------------------------------------------------------------
+            active                                   true
+            auth_key                                 XXX
+            lowest_acceptable_pn                     1
+            sak                                      XXX
+            salt                                     XXX
+            ssci                                     1
+            SAI_MACSEC_SA_ATTR_CURRENT_XPN           6661
+            SAI_MACSEC_SA_STAT_IN_PKTS_DELAYED       0
+            SAI_MACSEC_SA_STAT_IN_PKTS_INVALID       0
+            SAI_MACSEC_SA_STAT_IN_PKTS_LATE          0
+            SAI_MACSEC_SA_STAT_IN_PKTS_NOT_USING_SA  1
+            SAI_MACSEC_SA_STAT_IN_PKTS_NOT_VALID     0
+            SAI_MACSEC_SA_STAT_IN_PKTS_OK            8
+            SAI_MACSEC_SA_STAT_IN_PKTS_UNCHECKED     0
+            SAI_MACSEC_SA_STAT_IN_PKTS_UNUSED_SA     0
+            SAI_MACSEC_SA_STAT_OCTETS_ENCRYPTED      523517
+            SAI_MACSEC_SA_STAT_OCTETS_PROTECTED      0
+            ---------------------------------------  ----------------------------------------------------------------
+    '''
+    out = {'egress': {}, 'ingress': {}}
+    stats = None
+    reg = re.compile(r'(SAI_MACSEC.*?) *(\d+)')
+    for line in text.splitlines():
+        line = line.strip()
+
+        # Found the egress header, following stats will be for egress
+        if line.startswith("MACsec Egress SA"):
+            stats = 'egress'
+            continue
+        # Found the ingress header, following stats will be for ingress
+        elif line.startswith("MACsec Ingress SA"):
+            stats = 'ingress'
+            continue
+        # No header yet, so no stats coming
+        if not stats:
+            continue
+
+        found = reg.match(line)
+        if found:
+            out[stats].update({found.group(1): int(found.group(2))})
+    return out
+
+
+def get_macsec_counters(sonic_asic, port):
+    cmd = f"show macsec {port}"
+    output = sonic_asic.sonichost.command(cmd)["stdout"]
```

Comment:
This command is support at the Linecard level. We can get the output from duthost?